joschahenningsen/TUM-Live
# Add KaTeX

### Motivation and Context


- Fixes https://github.com/joschahenningsen/TUM-Live/issues/679. 

### Description

- Adds [KaTex](https://katex.org/) as a dependency
- Uses the [autorender extension]( https://katex.org/docs/autorender.html) to LaTex Code within chat messages.
- Adds [copy-tex](https://github.com/KaTeX/KaTeX/tree/main/contrib/copy-tex) to make rendered code copyable.
### Notes
- I tried to make it as performant as possibly by only including dependencies when needed. Furthermore, KaTeX code is only searched in the most elementary `<span>` that contains the chat message.
- I left full width equations enabled as they look surprisingly good out of the box.

Maybe this could be worth a server notification when this is deployed? I think otherwise people probably won't notice it.

### Steps for Testing

1. Log in.
2. Navigate to a livestream with chat enabled.
3. Post random equations such as `$\sum_{i=1}^{10} t_i$`.
4. Try to copy the rendered math and try to paste it somewhere. Check if it's your input.
5. Check if everything renders nicely without any noticeable delays or errors.

### Screenshots

![grafik](https://user-images.githubusercontent.com/30267166/198738538-cbbf170c-ed97-40e2-8b01-dd668ddb0bbd.png)
